File: src/node/child.js
import { ParentProcessBridge } from '../../lib/node_process_bridge/ProcessBridge.js'

let buffer = []
let parent = new ParentProcessBridge()

parent.on('end', () => {
    process.exit(0);
});

parent.on('receive', (data) => {
    buffer.push(data);
    if (buffer.length > 2) {
        second_stage();
    }
})

function second_stage() {
    let msgs = [
        `c1 ${buffer.join(' ')}`,
        'c2',
        'c3',
    ]
    for (const msg of msgs) {
        parent.send(msg)
        parent.send_err(msg)
    }
    setTimeout(() => {
        process.exit(12);
    }, 1000);
}    

File: src/python/child.py
import sys
import os
import time

python_folder = os.path.dirname(os.path.abspath(__file__))
src_folder = os.path.dirname(python_folder)
root_folder = os.path.dirname(src_folder)
sys.path.insert(0, f'{root_folder}/lib/python_process_bridge/')
from process_bridge import ParentProcess

parent = ParentProcess();

buf = []
buf.append(parent.receive())
buf.append(parent.receive())
buf.append(parent.receive())

parent.send(f'c1 {buf[0]} {buf[1]} {buf[2]}')
parent.send_err(f'c1 {buf[0]} {buf[1]} {buf[2]}')
parent.send('c2')
parent.send_err('c2')
parent.send('c3')
parent.send_err('c3')

time.sleep(1.0)

sys.exit(12)
